packages/lab/src/Blade/Blade.tsx

```diff
@@ -0,0 +1,307 @@
+import React, {
+  SyntheticEvent,
+  useCallback,
+  useMemo,
+  HTMLAttributes,
+  useEffect,
+} from "react";
+
+import {
+  ExtractNames,
+  HvBaseProps,
+  HvTypography,
+  HvTypographyVariants,
+  setId,
+  useControlled,
+  useDefaultProps,
+  useUniqueId,
+} from "@hitachivantara/uikit-react-core";
+
+import { staticClasses, useClasses } from "./Blade.styles";
+
+export { staticClasses as bladeClasses };
+
+export type HvBladeClasses = ExtractNames<typeof useClasses>;
+
+export interface HvBladeProps
+  extends HvBaseProps<HTMLDivElement, "onChange" | "children"> {
+  /**
+   * The content that will be rendered within the blade.
+   */
+  children: React.ReactNode;
+
+  /**
+   * The content of the blade's button.
+   */
+  label?: string | React.ReactNode;
+  /**
+   * Typography variant for the blade's button label.
+   */
+  labelVariant?: HvTypographyVariants;
+  /**
+   * Heading Level to apply to blade button.
+   *
+   * If 'undefined', the button will not have a header wrapper.
+   */
+  headingLevel?: 1 | 2 | 3 | 4 | 5 | 6;
+
+  /**
+   * A render function to customize the blade's button content.
+   *
+   * Useful for reflecting the blade's state in the button,
+   * e.g. disabled style, expanded indicator, etc..
+   *
+   * @returns
+   */
+  renderLabel?: (
+    props: Pick<
+      HvBladeProps,
+      "label" | "labelVariant" | "headingLevel" | "disabled" | "expanded"
+    >
+  ) => React.ReactNode;
```

Comment:
He doesn't have access to the props (the render function is out of the components closure), but he knows them naturally. But if reusing the same render function for multiple blades, it might be useful to receive them.
**However** that is not true for the `expanded`, as it is not the value of the prop `expanded` but the current state (when uncontrolled).

Confusing, I know... That is why I asked explicitly to review the API.

This could be the label prop, I just thought it could be nice to separate... E.g. I can pass a string as label and have a render function, shared with other blades, that uses that value. Again, maybe too confusing.

Why all of this? What I needed (for now) is the ability of having a complex label (not just a string, but a component)... But then I though: what if I need to reflect the state (e.g. displaying an "open" icon when expanded is true). Would I need to go to controlled mode, most likely in multiple blades, just for that? That is how the render function come to be.

Suggestions are welcome, including just removing it.

Reply:
That's right, `expanded` when uncontrolled would be a limitation. If we want to support uncontrolled, I see no better way than a render prop. 👍

Regarding the other _controlled_ props, I'm still not a fan, as the other props are redundant and extra maintenance (should a new prop (eg. `readOnly`) also be part of these props?)
Also, having both `label` and leveraging it in `renderLabel` is a bit strange (and the user already has access to it) and we lose type narrowing:
```tsx
label="shout this!"
renderLabel={({label}) => <div>{label.toUpperCase()}</div>}
// can't do this OOTB because it's 👆 a React.Node and not a string
```

Reply:
I'll merge it with the `label` prop and receiving only the expanded state.